scripts/components/UpdatePropertyForm.jsx

```diff
@@ -0,0 +1,146 @@
+var React = require('react');
+var moment = require('moment');
+var Api = require('../utils/Api.js');
+var MuiContextified = require('./MuiContextified.jsx');
+var mui = require('material-ui');
+var TextField = mui.TextField;
+var RaisedButton = mui.RaisedButton;
+var Dialog = mui.Dialog;
+var ImageSelector = require('./ImageSelector.jsx');
+var Label = require('./Label.jsx');
+var ErrorMessage = require('./ErrorMessage.jsx');
+
+var UpdatePropertyForm = React.createClass({
+  propTypes: {
+    propertyDetailsUpdated: React.PropTypes.func,
+    propertyID: React.PropTypes.number,
+  },
+
+  getInitialState() {
+    return {
+      tenantEmail: '', // User entered tenant email
+      image: '', // base64 encoding of the user selected image.
+      fileSizeError: '', // file size error message
+    }
+  },
+
+  onButtonClick() {
+    this.refs.dialog.show();
+  },
+
+  /**
+   * Event handler for capturing in the input field state on each keypress.
+   * @param  {String} field The identifier for the input field.
+   */
+  onChange(field, event) {
+    this.setState({ [field]: event.target.value });
+  },
+
+  /**
+   * Form submission event handler. Sends a request to the server to add the
+   * repair request, and updates the repair requests if successful.
+   */
+  onSubmit(event) {
+
+    // API call to update property details
+    var propertyId = this.props.propertyID;
+    Api.updatePropertyDetails({
+      data: {
+        tenantEmail: this.state.tenantEmail,
+        propertyId: this.props.propertyID,
+        image: this.state.image,
+      },
+      callback: (err, response) => {
+        if (err) {
+          return;
+        }
+
+        // Clear the form
+        this.setState({
+          tenantEmail: '',
+          image: '',
+        });
+
+        this.props.inspectionReportAdded();
```

Comment:
`this.props.inspectionReportAdded()` is not a property that is passed to the component. This means that the submit callback is not occurring, and the form is also not being dismissed.